cloudformation/slack-router-lambda.js

```diff
@@ -0,0 +1,44 @@
+const AWS = require('aws-sdk')
+
+function parseBody(event) {
+  const eventJSON = JSON.stringify(event.body, null, 2)
+  const eventBody = Buffer.from(eventJSON, 'base64').toString('ascii')
+  const bodyArray = eventBody.split('&')
+
+  const bodyObject = bodyArray.reduce((accumulator, currentValue) => {
+    const [key, value] = currentValue.split('=')
+
+    accumulator[key] = value
+
+    return accumulator
+  }, {})
+
+  return bodyObject
+}
+
+exports.handler = async (event) => {
+  const eventBody = parseBody(event)
+  const command = decodeURIComponent(eventBody.command).substring(1)
+
+  const params = {
+    Message: JSON.stringify(eventBody),
+    Subject: 'SNS from Slack Slash Command',
+    TopicArn: `arn:aws:sns:us-east-1:670261699094:${command}`,
+  }
+
+  try {
+    await new AWS.SNS().publish(params).promise()
+
+    return {
+      statusCode: 200,
+      body: 'Your message is being processed...',
+      headers: { 'Content-Type': 'application/json' },
+    }
+  } catch (error) {
+    return {
+      statusCode: 200,
+      body: 'Something went wrong with your request.',
+      headers: { 'Content-Type': 'application/json' },
+    }
```

Comment:
In the body, can we also tell them what went wrong? Also, let's return an error code for the statusCode, so that in the lambda logs, it registers as an error, not as a successful run. Can we not return a `Promise.reject` or a more explicit error, instead of returning the error as a successful response?

Reply:
80b15c26abeaaa7c5c4c2663294e430e190c7ac2

```
catch (error) {
    console.error(error) // Added to log error to CloudWatch

// this return block is to send back a message to Slack
    return {
      statusCode: 200,
      body:
        ':x: Something went wrong with your request. Please try again and if the error persists, post a message at <#C319P09PB>',
      headers: { 'Content-Type': 'application/json' },
    }
  }
```

I added a `console.error(error)` which will log the error in CloudWatch.

A 200 Status is needed for Slack to post the message. Otherwise Slack merely posts a generic message with no formatting.

See the difference here:
![Screen Shot 2020-07-14 at 19 02 18](https://user-images.githubusercontent.com/54427598/87395068-d1c8fb80-c604-11ea-853c-75a2b233294e.png)
The first error is what the code right now sends back (which is sent using the return block with the status 200) while the second one is what gets sent back if we throw an error (or do a Promise.reject), or send back a 500 status code.

@arunasank @ramyaragupathy let me know what you think we should do. Here is the [Slack docs](https://api.slack.com/interactivity/slash-commands#best_practices) as well and if you scroll up a bit there is a portion regarding **Sending error responses**

Reply:
Oh, sounds great @andria-hibe. I guess, since most of the users are non-developers/community members this definitely makes more sense. :+1: